packages/babel-parser/src/tokenizer/state.ts

```diff
@@ -23,40 +23,18 @@ type TopicContextState = {
   maxTopicIndex: null | 0;
 };
 
-const enum StateFlags {
-  None = 0,
-  Strict = 1 << 0,
-  maybeInArrowParameters = 1 << 1,
-  inType = 1 << 2,
-  noAnonFunctionType = 1 << 3,
-  hasFlowComment = 1 << 4,
-  isAmbientContext = 1 << 5,
-  inAbstractClass = 1 << 6,
-  inDisallowConditionalTypesContext = 1 << 7,
-  soloAwait = 1 << 8,
-  inFSharpPipelineDirectBody = 1 << 9,
-  canStartJSXElement = 1 << 10,
-  containsEsc = 1 << 11,
-}
-
 export const enum LoopLabelKind {
   Loop = 1,
   Switch = 2,
 }
 
+declare const bit: import("../../../../scripts/babel-plugin-bit-decorator/types.d.ts").BitDecorator<State>;
+
 export default class State {
-  flags: number = StateFlags.canStartJSXElement;
+  @bit.storage #flags: number;
```

Comment:
![image](https://github.com/babel/babel/assets/30521560/f0cfe2af-1beb-4a05-8dca-d3454e1fd78f)
Is there any special purpose for using private here?

Reply:
Nope, I'll revert to it being public. We can use a private on in Babel 8, since it minifies better.